sdks/router-sdk/src/entities/mixedRoute/route.ts

```diff
@@ -78,6 +92,10 @@ export class MixedRouteSDK<TInput extends Currency, TOutput extends Currency> {
     return this.pools[0].chainId
   }
 
+  private containsToken(pool: TPool, inputToken: Currency): boolean {
+    return (pool instanceof V4Pool && !pool.involvesToken(inputToken)) || !pool.involvesToken(inputToken.wrapped)
+  }
```

Comment:
why the !s?
if a pool involves token Y, this function is going to say it DOESNT contain token Y?

Reply:
like you have v3 pool [UNI, USDC]
input token USDC
`[USDC, UNI].involvesToken(USDC)` is true
so
`[USDC, UNI].containsToken(USDC)` is false ?

Reply:
oh i think i messed that up when i moved this to a private function 1 sec